src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -0,0 +1,260 @@
+        <div class="dropdown-placeholder dropdown-arrow"></div>
+      </div>
+      <div class="dropdown-content" v-if="showDropdown">
+        <div
+          v-for="optionName in Object.keys(selfCheckCourses)"
+          :key="optionName"
+          class="dropdown-content-item"
+          @click="addExistingCourse(optionName)"
+        >
+          <span>{{ optionName }}</span>
+        </div>
+        <div class="dropdown-content-item" @click="openCourseModal()">
+          <span>{{ '+ Add new course to schedule' }}</span>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+import { clickOutside } from '@/utilities';
+import store from '@/store';
+import { editSemesters } from '@/global-firestore-data';
+import { cornellCourseRosterCourseToFirebaseSemesterCourse } from '@/user-data-converter';
+
+import NewCourseModal from '@/components/Modals/NewCourse/NewCourseModal.vue';
+
+// enum to define seasons as integers in season order
+const SeasonsEnum = Object.freeze({
+  Winter: 0,
+  Spring: 1,
+  Summer: 2,
+  Fall: 3,
+});
+
+type Data = {
+  showDropdown: boolean;
+  isCourseModalOpen: boolean;
+};
+
+export default Vue.extend({
+  components: {
+    NewCourseModal,
+  },
+  props: {},
+  data(): Data {
+    return {
+      showDropdown: false,
+      isCourseModalOpen: false,
+    };
+  },
+  computed: {
+    selfCheckCourses(): Record<string, FirestoreSemesterCourse> {
+      // TODO - limit to courses that don't have requirements they are fulfilling
+      const courses: Record<string, FirestoreSemesterCourse> = {};
+      store.state.semesters.forEach(semester => {
+        semester.courses.forEach(course => {
+          courses[course.code] = course;
+        });
+      });
+      return courses;
+    },
+  },
+  directives: {
+    'click-outside': clickOutside,
+  },
+  methods: {
+    closeMenuIfOpen() {
+      this.showDropdown = false;
+    },
+    addExistingCourse(option: string) {
+      this.showDropdown = false;
+      this.$emit('addCourse', this.selfCheckCourses[option]);
+    },
+    addNewCourse(course: CornellCourseRosterCourse, season: FirestoreSemesterType, year: number) {
+      this.showDropdown = false;
+      const newCourse = cornellCourseRosterCourseToFirebaseSemesterCourse(course);
+      this.addCourseToSemester(season, year, newCourse);
+      this.$emit('addCourse', newCourse);
+    },
+    openCourseModal() {
+      this.isCourseModalOpen = true;
+    },
+    closeCourseModal() {
+      this.isCourseModalOpen = false;
+    },
+    checkCourseDuplicate(key: string) {
+      // if (this.courses) {
+      //   // @ts-ignore
+      //   this.$refs.modal.courseIsAddable = true;
+      //   this.courses.forEach(course => {
+      //     if (course.code === key) {
+      //       // @ts-ignore
+      //       this.$refs.modal.courseIsAddable = false;
+      //       this.$emit('open-caution-modal');
+      //     }
+      //   });
+      // }
+    },
```

Comment:
This code should be removed, and probably also the place where you copied it. I think it is pretty dead. The new course warning implementation doesn't require this code.